content_resolver/analyzer.py

```diff
@@ -1826,12 +1985,139 @@ def _resolve_srpm_using_root_log(self, srpm_id, arch, koji_session, koji_files_u
             raise KojiRootLogError("Could not get a root.log file")
 
         log("    Parsing the root.log file...")
-        directly_required_pkg_names = self._get_build_deps_from_a_root_log(root_log_contents)
+        directly_required_pkg_names = _get_build_deps_from_a_root_log_standalone(root_log_contents)
 
         log("    Done!")
         return directly_required_pkg_names
 
 
+    def _resolve_srpms_using_root_logs_parallel(self, pass_counter):
+        """
+        Parallel version of _resolve_srpms_using_root_logs using ProcessPoolExecutor.
+        This function is idempotent!
+        """
+        log("== Resolving SRPMs using root logs - pass {} (PARALLEL) ========".format(pass_counter))
+
+        # Collect work items (skip cached entries)
+        work_items = []
+        total_srpms_to_resolve = 0
+        
+        for koji_id in self.data["buildroot"]["koji_srpms"]:
+            koji_urls = self.data["buildroot"]["koji_urls"][koji_id]
+
+            # If the cache is empty, initialise it
+            if koji_id not in self.cache["root_log_deps"]["current"]:
+                self.cache["root_log_deps"]["current"][koji_id] = {}
+            if koji_id not in self.cache["root_log_deps"]["next"]:
+                self.cache["root_log_deps"]["next"][koji_id] = {}
+
+            for arch in self.data["buildroot"]["koji_srpms"][koji_id]:
```

Comment:
Why is it not necessary to initiate koji session(s) here as in the serial version?

Reply:
I believe it is because the serial version is having a single koji session for everything, and if there is more than one koji instance that is being looked at, it needs to have them both open and keep track of which package goes to what koji session.
In parallel, each package opens it's own koji session, so there is no need to keep track of them.